[alphajae11] iP

leejaegeun-sql opened the pull request:
### Chingu

> “Your mind is for having ideas, not holding them.” – David Allen (([source](https://dansilvestre.com/productivity-quotes)))

Chingu  frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

leejaegeun-sql committed:
Accomplished level-0

leejaegeun-sql committed:
Removed comments

leejaegeun-sql committed:
Accomplished Level-0

leejaegeun-sql committed:
Accomplished Level-1

leejaegeun-sql committed:
Accomplished Level-2

leejaegeun-sql committed:
Accomplished Level-3

leejaegeun-sql committed:
Accomplished Level-3

leejaegeun-sql committed:
Accomplished Level-4

leejaegeun-sql committed:
Accomplished A-TextUiTesting

leejaegeun-sql committed:
Accomplished A-TextUiTesting

leejaegeun-sql committed:
Accomplished Level-5

leejaegeun-sql committed:
Accomplished Level-6

leejaegeun-sql committed:
Accomplished Level-7

Commit:
Accomplished Level-8

Commit:
Attempted A-MoreOOP

leejaegeun-sql committed:
Attempted A-MOREOOP

leejaegeun-sql committed:
Accomplished A-MoreOOP

leejaegeun-sql committed:
Made 2 packages - duke and command

leejaegeun-sql committed:
Accomplished A-Packages

leejaegeun-sql committed:
Merge remote-tracking branch 'origin/add-gradle-support'

leejaegeun-sql committed:
Accomplished A-JUnit Tests

leejaegeun-sql committed:
Added to Level-8

acekhoon commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,156 @@
+        if (this.store.exists()) {
+            Scanner reader = new Scanner(store);
+            while (reader.hasNextLine()) {
+                if (reader.hasNext("T")) {
```
here some of the codes are repeating for handling T, D and E; maybe you can collapse into one single body and handle T, D and E thereafter; one other way is to use enums (that's what I did to make them shorter)

acekhoon commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,115 @@
+        this.description = description;
+    }
+
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
```
maybe the method can be more descriptive as mentioned in W4.6 & add header comments to explain what you did here

acekhoon commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,156 @@
+
+                } else if (temp instanceof Events) {
+                    System.out.println("OH");
+                    Events t = (Events) temp;
```
maybe for these methods, instead of type-casting I think you can use compile time type / run time type in java and just let everything to be casted to task class of checking individual object types

acekhoon commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,156 @@
+
+public class Storage {
+
+    public File store;
```
maybe the name could be more descriptive as explained in w4.6 (e.g. what to store)

acekhoon commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,156 @@
+                    String[] divided = data.split("\\|", 2);
+                    boolean isDone;
+                    String type = divided[0];
+                    data = divided[1];
```
again the name of variables could be more detailed and self-explanatory as mentioned above (e.g. data -> what data is in that variable?)

acekhoon commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,115 @@
+        if (description.isEmpty()) {
+            throw new DeadlineException("What deadline do you need to record?");
+        }
+        String [] divided = description.split("/by", 2);
```
here as well, variable names could be more self-explanatory (what information does "divide" contain?) Also, could be better if adding head comments

acekhoon commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,115 @@
+public class AddCommand extends Command {
+    private String taskType;
+
+    private String description;
```
here again, the variable names could be more self-explanatory and detailed!

leejaegeun-sql committed:
Added first JavaDocs of the majority of the classes and function

leejaegeun-sql committed:
Improved Coding Standard - 1st attempt

leejaegeun-sql committed:
Accomplished A-Jar

leejaegeun-sql committed:
Created FindCommand

leejaegeun-sql committed:
Cleaned up some unnecessary code

jamessinmaojun commented on src/main/java/duke/ToDos.java:
```diff
@@ -0,0 +1,25 @@
+/**
+ * This class is implementation of Task which contains description.
+ */
+public class ToDos extends Task {
```
I feel like you should name your class as `Todo` instead of `Todos` since this is technically not a collection of Todos. Also considering that you named your parent class `Task` instead of `Tasks` and `Deadline` instead of `Deadlines` I feel that you should stick to the singular naming convention.

jamessinmaojun commented on src/main/java/duke/Events.java:
```diff
@@ -0,0 +1,38 @@
+/**
+ * This class is implementation of Task which contains description, from and to.
+ */
+public class Events extends Task {
```
Similar to my comment on your `Todos` class, I feel like this class should be named `Event` instead of `Events`.

jamessinmaojun commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,78 @@
+        System.out.println("Here are the matching tasks in your list:");
+    }
+
+    public void markedDone(Task task) {
```
I understand why you named this in past tense but by Java coding convention your method should be named `markDone` instead.

jamessinmaojun commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,78 @@
+                "\t" + task.toString());
+    }
+
+    public void markedUndone(Task task) {
```
Same as above.

jamessinmaojun commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,115 @@
+package duke.command;
+
+import duke.*;
```
Should this be separated into different imports instead of `*`?

jamessinmaojun reviewed (commented):
Good job overall! Very neat organisation!

nobodyishappy commented on build.gradle:
```diff
@@ -0,0 +1,45 @@
+    dependsOn("distZip", "distTar")
+}
+
+run{
```
Should there be a space after the run?

nobodyishappy commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,49 @@
+package duke;
+
+import duke.command.*;
```
Should this be split up into different import statements?

nobodyishappy commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,166 @@
+ */
+public class Storage {
+
+    public File store;
```
Should this variable name be something like "file" instead as it better describes the variable?

nobodyishappy commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,166 @@
+                    String data = reader.nextLine();
+                    String[] divided = data.split("\\|", 2);
+                    boolean isDone;
+                    String type = divided[0];
```
Is this statement used in your code?

nobodyishappy commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,166 @@
+                if (reader.hasNext("T")) {
+                    String data = reader.nextLine();
+                    String[] divided = data.split("\\|", 2);
+                    boolean isDone;
```
Would this be better outside the if else statement instead?

nobodyishappy commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,166 @@
+                    if (temp.isDone) {
+                        isDone = "1";
+                    }
+                    to_record += "T" +" | " + isDone
```
Would this be better if it was in camel case instead?

nobodyishappy commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,46 @@
+    /**
+     * Creates an instance of Tasklist where there is no task in the file
+     */
+    public  TaskList() {
```
Should this have a single spacing between the public and name?

nobodyishappy commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,78 @@
+        System.out.println(LINE);
+    }
+
+    public void announceListing() {
```
Would showList be a better name for the function?

nobodyishappy reviewed (commented):
Some improvement to coding quality would be good but overall great job!!

leejaegeun-sql committed:
Added JavaFX side project trial in separate directory

leejaegeun-sql committed:
Accomplished Level-10

leejaegeun-sql committed:
parse in Parser is able to sort out the command based on the type of the command from the user
However, parse in Parser is unable to take care of the different possible errors in the command committed by the user.
In order to solve this issue, assert feature is used to cater for more possible specific errors commmited by the user. For example, assert feature checks if user wrongly puts todo as todos and it will return an error statement by stating that correct way of command is todo instead of todos.
Assertion can indicate a possible bug by the command from the user
Refer to this information about assertion: https://nus-cs2103-ay2324s2.github.io/website/schedule/week5/topics.html

leejaegeun-sql committed:
Parser, Storage, AddCommand class consist of many arrowheads and long functions
Therefore, there was a need to breakdown the functions to different levels of functions in order to remove long functions and arrowheads
Add new functions to Parser, Storage and AddCommand classes

leejaegeun-sql committed:
Add Priority feature into Doummi

Each task lacks the priority value so adding priority show its importance.

leejaegeun-sql committed:
Storage class: load the tasks from the list.txt

Storage class have some unintended mistake while loading by not recognized done task.

Trimming the status from the status collected from the list.txt to prevent this issue. For example, trim helps to get rid of the extra whitespace that causes the current error.

Using trim is suitable as it conveniently removes unnecssary whitespace in the status.

Refer to this website on how trim is used
https://www.w3schools.com/java/ref_string_trim.asp

leejaegeun-sql committed:
Relocate the Junit Test for EventTest

Necessary to comply with the file organisation

leejaegeun-sql committed:
Add Junit Test for Deadline Test

To comply with the deliverable requirement for Junit test.

leejaegeun-sql committed:
Rename the class as Todo

To comply with the Coding Standards for naming of classes

leejaegeun-sql committed:
Add Junit Test for ToDo Test

To comply with the deliverable requirements for Junit Test

leejaegeun-sql committed:
Change profile photos for user and chingu

To give more characteristics to Chingu

The previous photo was more like original duke.

leejaegeun-sql committed:
Add check split functions to AddCommand class

To comply with code quality standards

The processEvent and processDeadline is too long functions which need to be broken down.

leejaegeun-sql committed:
Merge pull request #2 from alphajae11/Merging-Branch

Add check split functions to AddCommand class

leejaegeun-sql committed:
Edit the import statements in some class

To comply with the coding standards (Avoid using *)

leejaegeun-sql committed:
Merge pull request #3 from alphajae11/branch-A-Assertions

Edit the import statements in some class

leejaegeun-sql committed:
Commit the current files that can create a Jar file

build.gradle can create a jar file now.

leejaegeun-sql committed:
Upload the first draft for User Guide of Chingu program

The User Guide can be found in the README.md in docs folder.

leejaegeun-sql committed:
Increase the size of the app

App was too small. It was difficult to fit in list of task which contains more than 3 tasks.

leejaegeun-sql committed:
Made changes to comply with code quality

leejaegeun-sql committed:
Make following changes to the source code

Fix the save functions to allow Jar File to automatically save all tasks when program ends

Added JavaDocs to the necessary public functions and classes

leejaegeun-sql committed:
Make following few changes to the source code

Cite references that are made in this project

Add JavaDocs to more public classes and functions

leejaegeun-sql committed:
Fix the 2 bugs

Variable name changes which affected the saving of the file.

FXML settings returned to original settings
